### ISSUE TITLE
Test that Testing and Validation are separate

### ISSUE THREAD

**Issue by miczed** [Hack3rz-Official/annotation-web-service#92]
- [x] write test or produce sufficient documentation for it (test that splitting data worked)
- [x] compute accuracy always on the same validation data? Maybe refactor the data splitting part and have a fixed validation dataset while we train the models with 100% training data?
- [x] Optimize training: stop training after loss convergence NOT strictly after 10 epochs
